rdkPlugins/Minidump/source/AnonymousFile.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2021 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "AnonymousFile.h"
+#include "Logging.h"
+
+#include <fcntl.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+// some platforms are missing the memfd headers
+// #include <linux/memfd.h>
+#if !defined(MFD_CLOEXEC)
+#  define MFD_CLOEXEC           0x0001U
+#endif
+#if !defined(MFD_ALLOW_SEALING)
+#  define MFD_ALLOW_SEALING     0x0002U
+#endif
+
+// for some reason the XiOne toolchain is build against old kernel headers
+// which doesn't have the memfd syscall
+#if !defined(SYS_memfd_create)
+#  if defined(__arm__)
+#    define SYS_memfd_create    385
+#  elif defined _MIPS_SIM
+#    if _MIPS_SIM == _MIPS_SIM_ABI32
+#      define SYS_memfd_create 4354
+#    endif
+#    if _MIPS_SIM == _MIPS_SIM_NABI32
+#      define SYS_memfd_create 6318
+#    endif
+#    if _MIPS_SIM == _MIPS_SIM_ABI64
+#      define SYS_memfd_create 5314
+#    endif
+#  endif
+#endif
+
+// glibc prior to version 2.27 didn't have a syscall wrapper for memfd_create(...)
+#if defined(__GLIBC__) && ((__GLIBC__ < 2) || ((__GLIBC__ >= 2) && (__GLIBC_MINOR__ < 27)))
+#include <syscall.h>
+static inline int memfd_create(const char *name, unsigned int flags)
+{
+    return syscall(SYS_memfd_create, name, flags);
+}
+#endif
+
+/**
+ * @brief Constructor - called when volatile file is about to be created
+ *
+ * @param[in]  size  size boundary of a volatile file
+ */
+AnonymousFile::AnonymousFile(size_t size) : mSize(size)
```

Comment:
`mFd` should be initialised to -1 here now you've added the `!= -1` check in `create()`